streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -829,6 +840,133 @@ public void shouldReadFutureProcessFileFormat() throws Exception {
         assertThat(directory.initializeProcessId(), equalTo(processId));
     }
 
+    @Test
+    public void shouldNotInitializeStandbyTasksWhenNoLocalState() {
+        initializeTasksForLocalState(new TaskId(0, 0), false);
+        assertFalse(directory.hasPendingTasks());
+    }
+
+    @Test
+    public void shouldInitializeStandbyTasksForLocalState() {
+        initializeTasksForLocalState(new TaskId(0, 0), true);
+        assertTrue(directory.hasPendingTasks());
+    }
+
+    @Test
+    public void shouldNotAssignPendingTasksWeDontHave() {
+        final TaskId taskId = new TaskId(0, 0);
+        initializeTasksForLocalState(taskId, false);
+        final Task task = directory.removePendingTask(taskId);
+        assertNull(task);
+    }
+
+    private class FakeStreamThread extends Thread {
+        private final TaskId taskId;
+        private final AtomicReference<Task> result;
+
+        private FakeStreamThread(final TaskId taskId, final AtomicReference<Task> result) {
+            this.taskId = taskId;
+            this.result = result;
+        }
+
+        @Override
+        public void run() {
+            result.set(directory.removePendingTask(taskId));
+        }
+    }
+
+    @Test
+    public void shouldAssignPendingTaskToStreamThread() throws InterruptedException {
+        final TaskId taskId = new TaskId(0, 0);
+
+        initializeTasksForLocalState(taskId, true);
+
+        // main thread owns the newly initialized tasks
+        assertThat(directory.lockOwner(taskId), is(Thread.currentThread()));
+
+        // spawn off a "fake" StreamThread, so we can verify the lock was updated to the correct thread
+        final AtomicReference<Task> result = new AtomicReference<>();
+        final Thread streamThread = new FakeStreamThread(taskId, result);
+        streamThread.start();
+        streamThread.join();
+        final Task task = result.get();
+
+        assertNotNull(task);
+        assertThat(task, instanceOf(StandbyTask.class));
+
+        // verify the owner of the task directory lock has been shifted over to our assigned StreamThread
+        assertThat(directory.lockOwner(taskId), is(instanceOf(FakeStreamThread.class)));
+    }
+
+    @Test
+    public void shouldClosePendingTasksOnDirectoryClose() {
+        final StateStore store = initializeTasksForLocalState(new TaskId(0, 0), true);
+
+        assertTrue(directory.hasPendingTasks());
+        assertTrue(store.isOpen());
+
+        directory.close();
+
+        assertFalse(directory.hasPendingTasks());
+        assertFalse(store.isOpen());
+    }
+
+    @Test
+    public void shouldClosePendingTasksOnAutoCleanUp() {
```

Comment:
Do we want to do it this way? I though this test should be `shoud[Not]ClosePendingTasksOnAutoCleanUp` ?